source/fundamentals/connection/tls.txt

```diff
@@ -0,0 +1,296 @@
+files.
+
+.. important::
+
+   For production use, your MongoDB deployment should use valid
+   certificates generated and signed by the same certificate authority.
+   For testing, you can use self-signed certificates.
+
+The following list describes the components that you need to establish
+a connection with TLS:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 30 70
+
+   * - TLS Component
+     - Description
+
+   * - Certificate Authority (CA)
+     - One or more certificate authorities to
+       trust when making a TLS connection.
+
+   * - Client Certificate
+     - A digital certificate and key that allow the server to verify the identity
+       of your application to establish an encrypted network connection.
+
+   * - Certificate Key
+     - The client certificate private key file. This key is often
+       included within the certificate file itself.
+
+   * - Passphrase
+     - The password to decrypt the client private key if it is encrypted.
+
+.. tip::
+   
+   To learn more about the PEM format, see the Wikipedia entry on
+   :wikipedia:`Privacy-Enhanced Mail <Privacy-Enhanced_Mail>`.
+
+.. _node-client-tls-connect:
+
+Reference Certificates in a Client
+----------------------------------
+
+You must reference your certificates in your ``MongoClientOptions``
+object so that the server can validate them before the client connects.
+You can reference your certificates in the following ways:
+
+- Create a ``SecureContext`` object to store certificates *(Recommended)*
+- Provide filepath strings that point to your certificates
+- Create ``Buffer`` objects to store certificates
+
+.. _node-tls-securecontext:
+
+Create a ``SecureContext`` Object to Store Certificates
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+We recommend that you use the ``secureContext`` option to configure
+your TLS connection. ``SecureContext`` objects are native to Node.js
+and allow you to keep all your TLS options in a single reusable object.
+
+To create a ``SecureContext`` object, you must first install the ``tls``
+package and use it in your application. Next, call the ``tls.createSecureContext()``
+method and pass the contents of your certificates in the options parameter.
+This method returns a ``SecureContext`` object that you can set as an option 
+in your ``MongoClientOptions`` object.
+
+The following code shows how to create a ``SecureContext`` object by
+calling the ``tls.createSecureContext()`` method and set the
+``secureContext`` client option:
+
+.. code-block:: js
+   :emphasize-lines: 2-6, 10
+   
+   // Create a SecureContext object
+   var sc = tls.createSecureContext({
+     ca: fs.readFileSync(`<path to CA certificate>`),
+     cert: fs.readFileSync(`<path to client certificate>`),
+     key: fs.readFileSync(`<path to client certificate>`),
+   });
+   
+   // Pass the SecureContext as a client option
+   const client = new MongoClient(uri, {
+     secureContext: sc,
+   });
```

Comment:
```suggestion
   const client = new MongoClient(uri, { tls: true, secureContext });
```

With the variable renamed this can be made more succinct